src/index.js

```diff
@@ -25,13 +26,15 @@ function WalletWatcher({ children }) {
       const walletAddress = accounts[0] ? accounts[0].toLowerCase() : undefined;
       const token = restoreToken({ walletAddress });
       const chainId = await window.ethereum.request({ method: 'eth_chainId' });
+      const contract = new Contract(address, abi, signer);
 
       updateSynthetix({
         walletAddress,
         token,
         provider,
         signer,
         chainId,
+        contract,
```

Comment:
As contract does not need any kind of special initialisation or async, we dont need it to be in the state. You can create `new ethers.Contract` right in the place where it is used. On the fly.
This would be better also because ethers contract has internal state (like connected signer) and it would not be a great idea to have a global shared instance of it. Instead, create Contract instance on demand for security and safety.

Reply:
Removed contract from global app state, as redundant and insecure approach, thx